cfme/fixtures/authentication.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+from time import sleep
+
+import cfme.utils.auth as authutil
+from cfme.utils.appliance import current_appliance
+from cfme.utils.log import logger
+
+
+def get_auth_settings(appliance):
+    """Grab the authentication settings from the UI form, popping the title widget content out"""
+    settings = {'auth_mode': appliance.server.authentication.auth_mode}
+    settings['auth_settings'] = appliance.server.authentication.auth_settings
+    settings['auth_settings'].pop('title')
+    return settings
+
+
+@pytest.fixture(scope='module')
+def amazon_auth_provider():
+    try:
+        return authutil.get_auth_crud('amazon')
+    except KeyError:
+        pytest.skip('amazon auth provider not found in auth_data.auth_providers, skipping test.')
+
+
+@pytest.fixture(scope='module')
+def setup_aws_auth_provider(appliance, amazon_auth_provider):
+    """Configure AWS IAM authentication mode"""
+    appliance.server.authentication.configure(auth_mode='amazon',
+                                              auth_provider=amazon_auth_provider)
+    yield
+
+    current_appliance.server.login_admin()
+    appliance.server.authentication.configure(auth_mode='database')
+
+
+@pytest.fixture(scope='function')
+def auth_provider(prov_key):
+        return authutil.get_auth_crud(prov_key)
+
+
+@pytest.fixture(scope='function')
+def auth_user_data(auth_provider, user_type):
+    try:
+        data = [user
+                for user in auth_provider.user_data
+                if user_type in user.user_types]
+        assert data
+    except (KeyError, AttributeError, AssertionError):
+        logger.exception('Exception fetching auth_user_data from yaml')
+        pytest.skip('No yaml data for auth_prov {} under "auth_data.test_data"'
+                    .format(auth_provider))
+    return data
+
+
+@pytest.yield_fixture(scope='function')
+def configure_auth(appliance, auth_mode, auth_provider, user_type, request):
+    if auth_mode.lower() != 'external':
+        appliance.server.authentication.configure(auth_mode=auth_mode,
+                                                  auth_provider=auth_provider,
+                                                  user_type=user_type)
+    elif auth_mode.lower() == 'external':  # extra explicit
+        if auth_provider.auth_type == 'freeipa':
+            appliance.configure_freeipa(auth_provider)
+            request.addfinalizer(appliance.disable_freeipa)
+        elif auth_provider.auth_type == 'openldaps':
+            appliance.configure_openldap(auth_provider)
+            request.addfinalizer(appliance.disable_openldap)
+
+    # Auth reconfigure is super buggy and sensitive to timing
+    sleep(30)
+    yield
+    # return to database auth config
+    appliance.server.authentication.configure(auth_mode='database')
```

Comment:
**QUESTION:** Is it possible that this will not always be the default (i.e. auth_mode='database)?   Would it be better to capture what the auth_mode was before hand and set it back to that?   
**OPTIONAL:** Consider making 'database' some sort of constant with a name like DEFAULT_AUTH_MODE (or however that would match our standards...maybe all caps is bad).

Reply:
This applies in a few other places I think.

Reply:
Yes, and that's a great suggestion. I'll adjust this accordingly.